src/main/scala/com/scalatsi/TypescriptType.scala

```diff
@@ -147,6 +147,8 @@ object TypescriptType {
   }
   case object TSVoid extends TypescriptType
 
+  case class TSFunction(arguments: List[(String, TypescriptType)], returnType: TypescriptType) extends TypescriptType
```

Comment:
A ListMap will allow both ordering and lookup by parameter name
```suggestion
  case class TSFunction(arguments: ListMap[String, TypescriptType], returnType: TypescriptType) extends TypescriptType
```